### PR TITLE
Only install binary packages on the second install pass.

### LINKED ISSUE #1482: Allow single-platform macOS apps
### What is the problem or limitation you are having?

With the addition of #1475, macOS apps are guaranteed to be universal, regardless of whether the wheels supplying binary components are universal.

However, there are legitimate use case for *not* producing universal binaries:
1. Concern over app size
2. Lack of availability of wheels on a given platform
3. A desire to drop x86_64 as a supported platform.

The last 2 points points in particular will only increase in frequency over time, as x86_64 macOS machines become increasingly uncommon.

Briefcase should allow an app to specify that it should *not* build a universal app.

### Describe the solution you'd like

A new option, `universal_build`, honored by the macOS backends. The default value is `True` for now; new apps templates should include an explicit value for this setting. At some point in the future, this default will be changed to `False`, and then the setting will be deprecated completely.

The current (post-#1475) behavior should continue if `universal_build==True`. If the user specifies `universal_build=False`, the older pre-#1475 pip install behavior should be restored (i.e., only wheels that match the current platform will be installed)

### Describe alternatives you've considered

Do nothing, and force all apps to be universal.

### Additional context

Updates to the [macOS App](https://github.com/beeware/briefcase-macOS-app-template/) and [macOS Xcode](https://github.com/beeware/briefcase-macOS-Xcode-template/) templates will also be required to set the new value explicitly.

### PR DESCRIPTION
**Pull request by freakboy3742**
More edge cases for macOS universal binary builds; this time, revealed by CI on [Python-support-testbed](https://github.com/beeware/Python-support-testbed/actions/runs/6452342478).

lru-dict only publishes x86_64 macOS wheels. When CI for Python-support-testbed is executed on x86_64, this results in an attempt to install a wheel for ARM64, which falls back to a source install, which then fails because the source install builds an x86_64 wheel.

The CI build fails on 3.8-3.10. It succeeds on 3.11 because the Python installed in CI is configured to produce universal wheels on compilation. 

CI for Python-support-testbed also fails for cffi on Python3.8, for the same reason.

This PR adds the `--only-binary :all:` option for secondary "other" arch install pass. This will cause the macOS build to fail if you're installing on x86-64 hardware, because there's no ARM64 or universal wheel available. This will require any project using lru-dict, cffi, or any other package without ARM64 binary wheels to provide those wheels.

`--only-binary :all` is a risky option to turn on for the *first* pass, because there are packages that only provide source wheels. This PR only enables `--only-binary` on the *second* pass, which is by definition only installing binary wheels for packages that didn't provide a universal wheel on the first pass.

To provide an exit hatch for apps that *don't* provide wheels, this PR also addresses #1482, adding the ability to make apps single-platform by setting `universal_build=false` in pyproject.toml. This requires an update to the Xcode template (beeware/briefcase-macOS-Xcode-template#27), and is assisted by (but does not *require*) a change to the base Briefcase template (beeware/briefcase-template#78). No change is needed to the macOS-app template; the stub binary is installed, and then stripped as part of the template unrolling process.

This will also require a PR to Python-support-testbed to provide the missing macOS wheels. If we land this PR, we can re-run CI on the main branch of Python-support-testbed to confirm that this fix works (i.e., the build will fail, but with an error about missing packages, rather than an error about binary merging); then a PR on that repo can add the missing wheels to get the CI passing again.

Fixes #1482
Refs beeware/briefcase-template#78
Refs beeware/briefcase-macOS-Xcode-template#27

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/briefcase/commands/create.py [hunk 1]
```diff
@@ -456,6 +457,9 @@ def _pip_install(
         :param requires: The list of requirements to install
         :param app_packages_path: The full path of the app_packages folder into which
             requirements should be installed.
+        :param include_deps: Should dependencies of the named requirements also be
+            installed?
+        :param only_binary: Should non-binary packages be installed?
         :param progress_message: The waitbar progress message to display to the user.
         :param pip_kwargs: Any additional keyword arguments to pass to the subprocess
             when invoking pip.
```
Comment (mhsmith) [inline src/briefcase/commands/create.py:465]:
```suggestion
        :param pip_kwargs: Any additional keyword arguments to pass to subprocess.run
            when invoking pip.
```

**Review by mhsmith — COMMENTED** [review]
> This will cause the macOS build to fail if you're installing on x86-64 hardware, because there's no ARM64 or universal wheel available. This will require any project using lru-dict, cffi, or any other package without ARM64 binary wheels to provide those wheels.

That could be a pretty big inconvenience, especially for a beginner, or a package that isn't easy to build. So I think this indicates that we need #1482 now after all. And when a universal build fails, we should show a message explaining how to do a single-platform build instead.

File: src/briefcase/commands/create.py [hunk 2]
```diff
@@ -476,13 +480,8 @@ def _pip_install(
...
                 ]
-                + (
-                    [
-                        "--no-deps",
-                    ]
-                    if not include_deps
-                    else []
-                )
+                + (["--no-deps"] if not include_deps else [])
+                + (["--only-binary", ":all:"] if only_binary else [])
```
[the remark this answers is not in the document]
Comment (freakboy3742) [reply src/briefcase/commands/create.py:484]: Yeah - I thought about that, and decided to continue down the "explicit argument" path; but in retrospect I think I agree that a cover-all "pip-args" is a better approach.

**Comment by freakboy3742** [comment]
> > This will cause the macOS build to fail if you're installing on x86-64 hardware, because there's no ARM64 or universal wheel available. This will require any project using lru-dict, cffi, or any other package without ARM64 binary wheels to provide those wheels.
> 
> That could be a pretty big inconvenience, especially for a beginner, or a package that isn't easy to build. So I think this indicates that we need #1482 now after all. And when a universal build fails, we should show a message explaining how to do a single-platform build instead.

Agreed. I'll add that into this PR before submitting for re-review.

**Review by mhsmith — APPROVED** [review]

**Pull request merged**